### ISSUE TITLE
[Bug/Feature Request]: add null as return type for nullable relations

### ISSUE THREAD

**Issue by mishavantol** [laravel-idea/plugin#396]
### Feature Description

Hi Adelf,

If a foreign constrained is nullable the result of calling the property can be null. It would be great if this is set correctly in the ide_helper_models file so static analysis can warn about possible null assignments.

For example, migration:
```php
    public function up(): void
    {
        Schema::create('countries', function (Blueprint $table) {
            $table->id();
            $table->timestamps();
            $table->foreignId('user_id')->nullable()->constrained();
        });
    }
```
Models\User.php:
```php
    public function country(): HasOne
    {
        return $this->hasOne(Country::class);
    }
```
now results into:
```php
    /**
     * @property int $id
     * @property string $name
     * ...
     * @property Country $country
     * ...
     */
    class User extends Model {}
```
Where it actually can be `@property Country|null $country`

**Comment by adelf** [comment]
Fixed. Thank you.

**Comment by mishavantol** [comment]
Thank you for the quick action and the helpful plugin!

**Comment by adelf** [comment]
It should be fixed by the 4.4.2 version.
